### ISSUE TITLE
Opening spot triggers "New login to Spotify" email

### ISSUE THREAD

**Issue by klingtnet** [xou816/spot#163]
**Describe the bug**

Opening spot triggers a "New login to Spotify" email.

**To Reproduce**

1. Just open spot.

**Expected behavior**

It should reuse its session or follow what the official client does.

**General information:**

 - Distribution: Arch Linux kernel 5.11.6
 - Installation method: AUR
 - Version: `0.1.10`
 - Device used: desktop

**Comment by Douile** [comment]
We could reduce the frequency of these emails by trying to use the token, instead of just the username and password every time spot is opened. I will look into it.

Relevant auto-login code:
- [Event triggerer](https://github.com/xou816/spot/blob/8f4186550ea12d43fe32e6136b64de887512c5ed/src/app/components/login/login_model.rs#L15-L23)
- [Player notifier](https://github.com/xou816/spot/blob/1d361b1c81514dc6b5e42846456859dbb1f326e9/src/app/components/player_notifier.rs#L38-L40)

This however might be impossible with librespot, I'm not entirely sure.

**Comment by xou816** [comment]
Yes, that was my original intent when storing the token along with username/password but AFAICT it is not possible to authenticate with that token in librespot. But I might have missed something -- afterall it's librespot that gives me this token... hm...

**Comment by xou816** [comment]
Oh and also -- the mails stop coming after a while, don't know why ;p

**Comment by alosarjos** [comment]
> Yes, that was my original intent when storing the token along with username/password but AFAICT it is not possible to authenticate with that token in librespot. But I might have missed something -- afterall it's librespot that gives us this token... hm...

Hmmm, this is in the code for sure...

https://github.com/librespot-org/librespot/blob/master/protocol/proto/authentication.proto#L25

Though it doesn't say anything about how to use it on the docs....

https://github.com/librespot-org/librespot/blob/master/docs/authentication.md

**Comment by graynk** [comment]
This still happens for me, I built from source from master a couple of days ago. Do I have to change something?

**Comment by xou816** [comment]
@nn1ks's patch should make it less frequent, but it might still happen and I don't think there is much else we can do :/ (I had not received any mail in a long while but I just had one today after Spotify's asked me to confirm my email)

**Comment by graynk** [comment]
Hmm. I'm switching between Spot and [Psst](https://github.com/jpochyla/psst), maybe that's the reason why I get them daily

**Comment by xou816** [comment]
Actually maybe Psst handles login better, doesn't seem to trigger emails! :thinking:

**Comment by nn1ks** [comment]
To clarify: My patch uses the access token which is only valid for one hour. So if you close Spot it could take anywhere between 0 and 60 minutes until the token is expired (depending on when it was last refreshed) and if it is expired you will still get an email the next time you open Spot.

I think  a solution to this could be to use the ["Authorization Code Flow"](https://developer.spotify.com/documentation/general/guides/authorization-guide/#authorization-code-flow) which in addition to an access token also returns a refresh token that is valid until it is manually revoked. The refresh token can then be used to request a new access token if it is expired. ([spotify-tui](https://github.com/Rigellute/spotify-tui) also uses this authentication method)

**Comment by xou816** [comment]
I don't know about spotify-tui, but I doubt Psst does the same -- it also goes through the undocumented Spotify endpoints that librespot uses to allow login in without opening a web browser. And yet it seems to work a bit better... I should have a closer look!
